### PR TITLE
refactor: IssuerTable based on brands rather than Keywords

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
Extracted from Zoe facet parameters (#1129, #718, #1116, in progress)

### CODE REVIEW

File: packages/zoe/src/zoe.js [hunk 1]
```diff
@@ -347,36 +349,33 @@ const makeZoe = (additionalEndowments = {}) => {
...
     const { issuerKeywordRecord } = instanceTable.get(instanceHandle);
+    // this method presumes that issuers have all been retrieved by this point
     sparseKeywords.forEach(keyword => {
-      const issuer = issuerKeywordRecord[keyword];
-      amountMathKeywordRecord[keyword] = issuerTable.get(issuer).amountMath;
+      const brand = issuerTable.brandFromIssuer(issuerKeywordRecord[keyword]);
+      amountMathKeywordRecord[keyword] = issuerTable.get(brand).amountMath;
     });
-    return harden(amountMathKeywordRecord);
+    return amountMathKeywordRecord;
```
Comment (katelynsills) [inline packages/zoe/src/zoe.js]: should we keep the harden?

File: packages/zoe/src/state.js [hunk 2]
```diff
@@ -96,66 +96,79 @@ const makeOfferTable = () => {
...
 
   const makeCustomMethods = table => {
     const issuersInProgress = makeStore();
+    const issuerToBrand = makeStore();
+
+    // We can't be sure we can build the table entry soon enough that the first
+    // caller will get the actual data, so we start by saving a promise in the
+    // inProgress table, and once we have the Issuer, build the record, fill in
+    // the table, and resolve the promise.
+    function buildTableEntryAndPlaceHolder(issuer) {
```
Comment (katelynsills) [inline packages/zoe/src/state.js:121]: Pulling this out into a separate function is a great idea

**Review by katelynsills — CHANGES_REQUESTED** [review]
I think the core of this looks very good. There are some extraneous things and one addition to the API that I want to remove, but otherwise looks ready to merge

File: packages/zoe/src/zoe.js [hunk 3]
```diff
@@ -616,7 +615,9 @@ const makeZoe = (additionalEndowments = {}) => {
         );
       },
       getInstanceRecord: () => instanceTable.get(instanceHandle),
-      getIssuerRecord: issuer => removePurse(issuerTable.get(issuer)),
+      getAmountMathForBrand,
```
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply packages/zoe/src/zoe.js]: done

File: packages/zoe/src/zoe.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/zoe.js]: done

File: package.json [hunk 4]
```diff
@@ -65,5 +65,8 @@
     "lint-check": "yarn workspaces run lint-check",
     "test": "yarn workspaces run test",
     "build": "yarn workspaces run build"
+  },
+  "dependencies": {
+    "@agoric/store": "^0.1.2"
```
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply package.json]: done

**Comment by Chris-Hibbert** [comment]
Sorry, just saw some more comments. I'll address them now.

**Review by katelynsills — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**